src/pages/JoinEvent/JoinEvent.tsx

```diff
@@ -0,0 +1,216 @@
+import "./JoinEvent.scss";
+import { useState, FormEvent, ChangeEvent, useEffect } from "react";
+import { Link, useParams, useNavigate } from "react-router-dom";
+import { doc, getDoc, setDoc } from "firebase/firestore";
+import { db } from "../../Firebase/FirebaseConfig";
+import DashboardNavbar from "../../components/DashboardNavbar/DashboardNavbar";
+import { useAuth } from "../../context/AuthContext";
+
+const JoinEvent = () => {
+  const { eventId } = useParams();
+  const navigate = useNavigate();
+  const { currentUser } = useAuth();
+
+  const [name, setName] = useState("");
+  const [location, setLocation] = useState("");
+  const [timezone, setTimezone] = useState("(GMT-0700) Canada (Vancouver) Time");
+  const [skillLevel, setSkillLevel] = useState("Beginner");
+  const [discipline, setDiscipline] = useState("Developer");
+  const [goal, setGoal] = useState("");
+  const [strength, setStrength] = useState("");
+  const [weakness, setWeakness] = useState("");
+  const [eventData, setEventData] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchEventData = async () => {
+      try {
+        const docRef = doc(db, "In_hackathonEvents", eventId);
+        const docSnap = await getDoc(docRef);
+        if (docSnap.exists()) {
+          setEventData(docSnap.data());
+        } else {
+          console.log("No such document!");
+          navigate("/hackathons");
+        }
+      } catch (error) {
+        console.error("Error fetching event data: ", error);
+        navigate("/hackathons");
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchEventData();
+  }, [eventId, navigate]);
+
+  const handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (!currentUser || !eventId) return;
+
+    const participantData = {
+      name,
+      location,
+      timezone,
+      skillLevel,
+      discipline,
+      goal,
+      strength,
+      weakness
+    };
+
+    try {
+      const eventRef = doc(db, "In_hackathonParticipantData", eventId);
+      const eventSnap = await getDoc(eventRef);
+
+      if (eventSnap.exists() && eventSnap.data().participants?.[currentUser.uid]) {
```

Comment:
you can take out participants and have it as eventSnap.data()?.[currentUser.uid]